docs/guides/features/ipsec-offload.md

```diff
@@ -0,0 +1,138 @@
+# IPsec Offload
+
+The Inline IPsec offload feature uses the
+[IPDK Infrastructure Application Interface](https://ipdk.io/documentation/Interfaces/InfraApp/)
+to enable cryptographically-secure data traffic. The IPsec control plane
+(IKE protocol) is offloaded, thus avoiding specialized drivers or control
+planes burdening compute instances. Further details are available on the
+[IPDK Inline Acceleration - IPsec](https://ipdk.io/documentation/Recipes/InlineIPsec/)
+page.
+
+This feature works in conjunction with the
+[IPsec Recipe](https://github.com/ipdk-io/ipsec-recipe), which provides
+strongSwan plugin code to configure and program control messages into the
+target.
+
+The IPsec offload feature is only supported on the Intel&reg; IPU E2100 target.
+
+## Feature overview
+
+The `infrap4d` process provides the gRPC server-side support for P4RT and
+gNMI messages. strongSwan acts as the gRPC client in this context.
+You can also use other clients that implement the IKE stack. The
+client uses P4RT to program the Security Policy Database (SPD), and it
+uses gNMI to configure Security Association Database (SAD) entries, which
+includes encryption keys, re-keying etc.
+
+The [openconfig-ipsec-offload](https://github.com/ipdk-io/openconfig-public/blob/master/release/models/ipsec/openconfig-ipsec-offload.yang)
+YANG model is used to configure IPsec offload.
+
+## Enabling IPsec
+
+Follow the sequence of steps listed below to enable IPsec functionality.
+
+### Load crypto package
+
+Follow the instructions in [Deploying P4 programs](/guides/es2k/deploying-p4-programs.md)
+to load the hardware FXP pipeline with the crypto package.
+
+### Configure and run infrap4d
+
+The OpenConfig portion of IPsec Offload functionality is enabled by using fixed
+functions configuration. Follow the instructions in [Running infrap4d](/guides/es2k/running-infrap4d.md)
```

Comment:
To be able able to program Security Association Database (SAD) entries using gNMI, enable fixed function support in infrap4d. 

Modify /usr/share/stratum/es2k/es2k_skip_p4.conf to enable fixed function.

Also, it's not clear where this block needs to be added in conf file

Reply:
Corrected the text with recommendation.

The block where it needs to be added should be evident when someone looks at the file, since there is a fixed_function block already present (and is blank by default)